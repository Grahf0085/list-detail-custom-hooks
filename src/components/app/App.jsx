import React from 'react';
import {
  BrowserRouter as Router,
  Route,
  Switch,
  Redirect
} from 'react-router-dom';
import ToonList from './ToonList';
import ToonDetailPage from './ToonDetailPage';
import Header from './Header';
import './app.css';

export default function App() {
  return (<section>
    <Router>
      <Header />
      <Switch>
        <Route exact path="/" component={ToonList} />
        <Route exact path="/character/:id" component={ToonDetailPage} />
        <Redirect to="/" />
      </Switch>
    </Router>
  </section>
  );
}
