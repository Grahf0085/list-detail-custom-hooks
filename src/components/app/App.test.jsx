import React from 'react';
import { render, screen, waitFor } from '@testing-library/react';
import App from './App';
import { MemoryRouter } from 'react-router-dom';

describe('App component', () => {
  it('displays a list of characters on the home page', async () => {
    render(
      <MemoryRouter initialEntries={['/']}>
        <App />
      </MemoryRouter>
    );

    screen.getByText('Loading List...');

    const ul = await screen.findByRole('list');
    expect(ul).not.toBeEmptyDOMElement();
  });

  it('displays a characters details', async () => {
    render(
      <MemoryRouter initialEntries={['/character/1']}>
        <App />
      </MemoryRouter>
    );

    screen.getByText('Loading List...');
    waitFor(async () => {
      const section = await screen.findByRole('details'); 
      await screen.findByText('Alive');
      await screen.findByText('Male');
      await screen.findByText('Rick Sanchez');
      expect(section).not.toBeEmptyDOMElement();
    }, 1000);


  });
});
