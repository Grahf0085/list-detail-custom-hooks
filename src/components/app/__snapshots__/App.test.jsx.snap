// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component renders App 1`] = `
<DocumentFragment>
  <section>
    <header>
      <h1>
        Rick and Morty Toons 4.0132432487398503
      </h1>
      <a
        href="/"
      >
        <button>
          Home
        </button>
      </a>
    </header>
    <h1>
      Loading...
    </h1>
  </section>
</DocumentFragment>
`;
